### PR TITLE
[Compute] Fix a vmss update, disk attach/detach, extension set/delete issue

### PR DESCRIPTION
**Pull request by qwordy**
**Description**  

Resolve https://github.com/Azure/azure-cli/issues/14924
Resolve https://github.com/Azure/azure-cli/issues/15058

Support partial update of vmss to avoid unnecessary permission check.

How to try it. https://github.com/Azure/azure-cli/blob/dev/doc/try_new_features_before_release.md

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
vmss

File: src/azure-cli/azure/cli/command_modules/vm/commands.py [hunk 1]
```diff
@@ -412,7 +412,7 @@ def load_command_table(self, _):
         g.command('simulate-eviction', 'simulate_eviction', command_type=compute_vmss_vm_sdk, min_api='2019-12-01')
         g.custom_command('start', 'start_vmss', supports_no_wait=True)
         g.custom_command('stop', 'stop_vmss', supports_no_wait=True, validator=process_vm_vmss_stop)
-        g.generic_update_command('update', getter_name='get_vmss', setter_name='update_vmss', supports_no_wait=True, command_type=compute_custom, validator=validate_vmss_update_namespace)
+        g.generic_update_command('update', getter_name='get_vmss_modified', setter_name='update_vmss', supports_no_wait=True, command_type=compute_custom, validator=validate_vmss_update_namespace)
```
Comment (arrownj) [inline src/azure-cli/azure/cli/command_modules/vm/commands.py:415]: This is a very special case, it's better to add a standalone test for it.

Comment (arrownj) [reply src/azure-cli/azure/cli/command_modules/vm/commands.py:415]: Considering it needs to update the permission of user in a subscription, I'm OK to skip the test. Let's make sure the test is done in local.

**Review by arrownj — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/vm/commands.py (hunk 1 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/commands.py:415]: I test it manually

Comment (yungezz) [reply src/azure-cli/azure/cli/command_modules/vm/commands.py:415]: long term we need to switch to patch mode to completely resolve the issue. I'll file an issue to track it.

**Comment by qwordy** [comment]
I can't merge it now. It breaks a successfully test case. Let's wait for VMSS service team to support not setting image reference.

**Comment by yungezz** [comment]
service side ready, and tested the change works.

**Pull request merged**